### ISSUE TITLE
Stateless enviroments

### ISSUE THREAD

**Issue by AlexandraVolokhova** [alexhernandezgarcia/gflownet#334]
There're several methods in the environments which work with states (e.g. `state2policy`, `get_parents`) and either take state as an input or use internal state of the enrolment `self.state`. This was a source of bugs in the past (when an env was supposed to transform input states but is was transforming the internal state or vice versa)

Proposal (by @carriepl ):
Alternative approach would be to get rid of the internal state of the environment and require input state for all methods where the state is needed.

**Comment by alexhernandezgarcia** [comment]
I think I would like to hear more arguments for this change. If we make the environment stateless, then we should also get rid of `self.done` and pass it to all methods too, and probably `self.n_actions`, etc. It should be all or nothing, no?

**Comment by AlexandraVolokhova** [comment]
@carriepl could you elaborate on that? I'm not sure I fully understood your suggestion to be able to address Alex's question

**Comment by josephdviviano** [comment]
I would second this change

Joseph (Mobile)

**Comment by alexhernandezgarcia** [comment]
I'm not boldly against it, but it's a huge change that would require changing many methods in _all_ environments and all their units tests. Even if I wanted to have this change done in theory, I don't see myself doing it right now.

**Comment by josephdviviano** [comment]
Yes it makes sense to deprioritize it relative to the effort involved
(unless it’s possible to support both simultaneously)

Joseph (Mobile)
